storage-proofs/src/fr32.rs

```diff
@@ -34,6 +35,27 @@ pub fn bytes_into_fr<E: Engine>(bytes: &[u8]) -> Result<E::Fr> {
     E::Fr::from_repr(fr_repr).map_err(|_| Error::BadFrBytes)
 }
 
+#[inline]
+pub fn bytes_into_fr_repr_safe(r: &[u8]) -> FrRepr {
+    debug_assert!(r.len() == 32);
+
+    let repr: [u64; 4] = [
+        LittleEndian::read_u64(&r[0..8]),
+        LittleEndian::read_u64(&r[8..16]),
+        LittleEndian::read_u64(&r[16..24]),
+        u64::from(r[31] & 0b0011_1111) << 56
+            | u64::from(r[30]) << 48
+            | u64::from(r[29]) << 40
+            | u64::from(r[28]) << 32
+            | u64::from(r[27]) << 24
+            | u64::from(r[26]) << 16
+            | u64::from(r[25]) << 8
+            | u64::from(r[24]),
+    ];
```

Comment:
Not following this format, why do we force a little endian ordering (`LittleEndian::read_u64`) here? Is this the way the hasher will always output data?

Reply:
this is just doing what we were doing before, we need a canoncial way to go from [u8; 32] to [u64; 4] (the internal representation of an Fr element. And there are roughly to ways, one is little endian one is big, and we settled on little endian a while ago